### PR TITLE
Amazon Linux 2 on AWS

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
Amazon Linux 2 on AWS

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Support Amazon Linux 2 on AWS
```

### CODE REVIEW

File: pkg/userdata/amzn2/provider.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	if err != nil {
+		return "", fmt.Errorf("failed to get provider config: %w", err)
+	}
+
+	if pconfig.OverwriteCloudConfig != nil {
+		req.CloudConfig = *pconfig.OverwriteCloudConfig
+	}
+
+	if pconfig.Network != nil {
+		return "", errors.New("static IP config is not supported with CentOS")
```
Comment (moadqassem) [inline pkg/userdata/amzn2/provider.go:61]: Is this relevant to the amzn linux os as well?

File: cmd/userdata/amzn2/main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	"github.com/kubermatic/machine-controller/pkg/userdata/amzn2"
+	userdataplugin "github.com/kubermatic/machine-controller/pkg/userdata/plugin"
+)
+
+func main() {
+	// Parse flags.
+	var debug bool
+
+	flag.BoolVar(&debug, "debug", false, "Switch for enabling the plugin debugging")
```
[the remark this answers is not in the document]
Comment (kron4eg) [reply cmd/userdata/amzn2/main.go:36]: yes

**Comment by kron4eg** [comment]
> /test pull-machine-controller-e2e-aws-amzn2

this one has been removed right after adding since it's already included in `pull-machine-controller-e2e-aws`
